pkg/test/framework/components/istio/operator.go

```diff
@@ -137,21 +137,25 @@ func (i *operatorComponent) IngressFor(cluster cluster.Cluster) ingress.Instance
 	return i.CustomIngressFor(cluster, defaultIngressServiceName, defaultIngressIstioLabel)
 }
 
-func (i *operatorComponent) CustomIngressFor(cluster cluster.Cluster, serviceName, istioLabel string) ingress.Instance {
+func (i *operatorComponent) CustomIngressFor(c cluster.Cluster, serviceName, istioLabel string) ingress.Instance {
 	i.mu.Lock()
 	defer i.mu.Unlock()
-	if i.ingress[cluster.Name()] == nil {
-		i.ingress[cluster.Name()] = map[string]ingress.Instance{}
+	if c.Kind() != cluster.Kubernetes {
```

Comment:
Is this something we can just always do? Maybe a comment will help to clarify.

Reply:
I can't think of a good way to push this down further. 
If we make all k8s calls to the non-kube cluster go through to the kube cluster, test-writers could easily apply configuration to the same cluster twice and  possibly with different values. I'd rather than run into the same error I did, then add `ctx.Clusters().Kube()`. 


Either that or move `ExtendedClient` out of the `Cluster` interface, and force type-coversions but that will be a very widespread change.

Reply:
ok sg ... ultimately, I think we'll want to give the cluster interface a facelift similar to what you suggest. But we can kick the can on that for now.